### PR TITLE
Improve visual site

### PR DESCRIPTION
**Pull request by 1killerbehindyou1**
Implemented:
- fixed problem with file open and save
- added sliders
- fix problems with arguments in QML

### CODE REVIEW

File: BasicGUI/QMLSources/TransformatedPreview.qml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import QtQuick.Window 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.3
+import QtQuick.Dialogs 1.3
+import Main 1.0
+
+Control
+{ 
+    
+}
```
Comment (Nutek) [inline BasicGUI/QMLSources/TransformatedPreview.qml:11]: If this is a working state which is not necessary for this delivery then please keep this file only in local history or separate branch.

**Review by Nutek — APPROVED** [review]

File: BasicGUI/FileManager.h [hunk 2]
```diff
@@ -14,21 +14,25 @@
...
 public:
-    ImageViewer(QQuickItem* parent = 0);
+    FileManager(QObject* parent = 0);
+    QPixmap* getPixmapPointer();
 
-    void paint(QPainter* painter) override;
-    Q_INVOKABLE bool setPixMap(const QUrl& path);
+    QPixmap m_pixmap;
+    QPixmap m_transformed;
+    QString m_path;
```
[the remark this answers is not in the document]
Comment (1killerbehindyou1) [reply BasicGUI/FileManager.h:29]: Corrected

File: BasicGUI/Interpolator.h [hunk 3]
```diff
@@ -32,8 +31,7 @@ class Interpolator : public QObject
 
     QColor interpolatorSetLedColor(QVector<QPointF> vector_points);
     QVector<QPointF> interpolatorTransform(Transform transform, QRect rect);
-
-    QPixmap inerpolator_pixmap;
+    QPixmap* inerpolator_pixmap = 0;
```
[the remark this answers is not in the document]
Comment (1killerbehindyou1) [reply BasicGUI/Interpolator.h:34]: Corrected.

File: BasicGUI/LedRuler.cpp [hunk 4]
```diff
@@ -10,55 +10,66 @@
...
-        "Mentoring_QT_project\\Spectral_Display_GUI\\Pictures\\BITMAPA.png");
-    m_number_of_leds = 3;
-    m_rotation = 90;
-    m_spacing = 10;
-    m_size = 3;
+
+    m_interpolator.inerpolator_pixmap = nullptr;
+    m_number_of_leds = 30;
+    m_rotation = 1;
+    m_size = 1;
```
[the remark this answers is not in the document]
Comment (1killerbehindyou1) [reply BasicGUI/LedRuler.cpp:17]: Corrected

File: BasicGUI/LedRuler.cpp [hunk 5]
```diff
@@ -10,55 +10,66 @@
...
 {
-    m_interpolator.inerpolator_pixmap.load(
-        "C:\\Users\\mplesniak\\Desktop\\1. "
-        "Mentoring_QT_project\\Spectral_Display_GUI\\Pictures\\BITMAPA.png");
-    m_number_of_leds = 3;
-    m_rotation = 90;
-    m_spacing = 10;
-    m_size = 3;
+
+    m_interpolator.inerpolator_pixmap = nullptr;
```
[the remark this answers is not in the document]
Comment (1killerbehindyou1) [reply BasicGUI/LedRuler.cpp:14]: Corrected.

File: BasicGUI/QMLSources/TransformatedPreview.qml (hunk 1 above)
Comment (1killerbehindyou1) [reply BasicGUI/QMLSources/TransformatedPreview.qml:11]: Corrected

File: BasicGUI/QMLSources/qml.qrc [hunk 6]
```diff
@@ -4,6 +4,6 @@
         <file>RenderPanel.qml</file>
         <file>ControlPanel.qml</file>
         <file>DataInput.qml</file>
-        <file>ImagePanel.qml</file>
+        <file>TransformatedPreview.qml</file>
```
[the remark this answers is not in the document]
Comment (1killerbehindyou1) [reply BasicGUI/QMLSources/qml.qrc:7]: Corrected

File: BasicGUI/Interpolator.cpp [hunk 7]
```diff
@@ -10,7 +10,10 @@ Interpolator::Interpolator(QObject* parent) : QObject(parent) {}
 //////////////////////////////////////////////////////////////////
 QColor Interpolator::interpolatorSetLedColor(QVector<QPointF> vector_points)
 {
-    QImage image = inerpolator_pixmap.toImage();
+    QImage image{};
+    if (inerpolator_pixmap != nullptr)
+        image = inerpolator_pixmap->toImage();
+
```
[the remark this answers is not in the document]
Comment (1killerbehindyou1) [reply BasicGUI/Interpolator.cpp:16]: Corrected.

File: BasicGUI/QMLSources/main.qml [hunk 8]
```diff
@@ -21,52 +24,88 @@ ApplicationWindow
...
+                    drawing.saveImage();
+                }
+            }
         }
         Menu 
         {
-            title: qsTr("&Help")
-            Action { text: qsTr("&About") }
+            title: qsTr("&View")
+            Action
```
[the remark this answers is not in the document]
Comment (Nutek) [reply BasicGUI/QMLSources/main.qml:46]: Redirected to issue #10

**Review by Nutek — APPROVED** [review]

**Pull request merged**